### ISSUE TITLE
Write an integration test for HDFS

### ISSUE THREAD

**Issue by foxish** [apache-spark-on-k8s/spark-integration#10]
This would be a test to check that read from HDFS works. @kimoonkim, your HDFS can be launched on a k8s cluster for this test. Kerberized cluster is not needed. It could target the GCE_TEST_BACKEND to avoid taxing the minikube instance. This will ensure read/write from/to HDFS works as expected.
We only need this **after** https://github.com/apache/spark/pull/19954 is merged.

cc @ifilonenko

**Comment by foxish** [comment]
Yinan - HDFS should just work after that PR correct?

**Comment by liyinan926** [comment]
Yes, it should work for remote dependencies on both non-secured and secured HDFS clusters.

**Comment by foxish** [comment]
Isn't secured HDFS support still in our fork and not upstreamed yet? The KDC flow, refreshing DTs etc hasn't made it into the upstream PR IIRC.

**Comment by kimoonkim** [comment]
Secure HDFS support is still in our fork.

**Comment by liyinan926** [comment]
Actually to use a remote dependency on a secured HDFS cluster, one doesn't even need the work in our fork. All they need is mount a DT into the init-container, set environment variable `HADOOP_TOKEN_FILE_LOCATION` to point to the DT file, and the init-container should be able to download the file.

**Comment by foxish** [comment]
Hmm.. fair point - I guess the issue with that is that once the DT expires, one cannot re-fetch that dependency, which might be a deal-breaker for long running jobs. My thinking here is that we add the requisite logic to support it completely and then in 2.4, announce it. In 2.3, we can say full support for HDFS clusters with Kerberos. 

Maybe a small blurb can be added about the point you made regarding limited support for Kerberized clusters in 2.3 documentation - not sure if we should do that, let's discuss more.

**Comment by foxish** [comment]
Thinking that we should test both dependency on HDFS and data on HDFS. I think the latter will be more common - most people have large amounts of data on HDFS that they'd want to run a job against - so that's an interesting use-case.

**Comment by ifilonenko** [comment]
I have a hadoop-base image to test basic hdfs that lives on PR 414. I'll add that test after porting over 521.
